src/ORM/Query.php

```diff
@@ -1285,6 +1300,20 @@ public function insert(array $columns, array $types = [])
         return parent::insert($columns, $types);
     }
 
+    /**
+     * Returns a new Query that has automatic field aliasing disabled.
+     *
+     * @param \Cake\ORM\Table $table The table this query is starting on
+     * @return self
+     */
+    public function subquery(Table $table): self
+    {
+        $query = new self($this->getConnection(), $table);
```

Comment:
`static` instead of `self` perhaps?

Reply:
`static` should be ok.

Reply:
On second thought, other code explicitly creates `Database\Query` so maybe this should explicitly create `ORM\Query` instances.

Reply:
Still. If you ever needed to override on plugin or project level, it would not be easy to overcome this typehint then, would it? The static is just a bit more lenient here while it will provide the same functionality out of the box.